### ISSUE TITLE
False negatives when using variable named `fit` (`no-focused-tests` & `no-test-prefixes`)

### ISSUE THREAD

**Issue by marcohamersma** [jest-community/eslint-plugin-jest#416]
Heya,
I have a module called `fit` which is being imported in a non-test file. However, this plugin seems to think that this file has a test in it (because it searches for "`it`" ?) and throws two errors:

```js
const { width, height, scale } = fit(
  this.props.frameSize, container, { apply: false }
);
```  

![Unexpected focused test  jest/no-focused-tests
Use "it.only" instead    jest/no-test-prefixes](https://user-images.githubusercontent.com/577144/65429453-1a14f400-de0e-11e9-92fb-b8f3fb5f99f5.png)

I'm guessing this is not supposed to happen, so I wanted to report it.

**Comment by jeysal** [comment]
> because it searches for "`it`"

It also searches for `fit`, which is an API equiv. to `it.only`.
But yeah if `fit` is imported from somewhere else these warnings should not be shown.

**Comment by SimenB** [comment]
We should check it's a global, yeah

**Comment by G-Rath** [comment]
I feel like this might have the same solution as #209 
iirc, it's actually a lot of work to check if something is imported globally every time (but I might be thinking of variables)?

@SimenB what rules already do this?

**Comment by SimenB** [comment]
>@SimenB what rules already do this?

Any rule using `scopeHasLocalReference`

**Comment by G-Rath** [comment]
> >@SimenB what rules already do this?
> Any rule using scopeHasLocalReference

Yeah that is what I was thinking of.

I don't think it'll be a performance killer, but it always struck me as being a decent amount of work relative to everything else, so it'll be interesting to see if it has any noticeable effect if we start calling it for literally every rule multiple times over.

The main optimisation I'm thinking about is reusing the results of `collectReferences`.

I'll have a look to implement this when I get some free time.

**Comment by marcohamersma** [comment]
I guess part of the problem might also be that it's looking inside a file that wouldn't normally be loaded by jest. By default jest seems to only load files ending in `.test.js`, `.spec.js`, etc…, so maybe this behaviour can be mirrored? 

Might not be the ideal solution, but it would probably also lower the probability of errors like this?

**Comment by SimenB** [comment]
@marcohamersma people can specify whatever pattern they want, so I don't think this plugin should care about filenames. Maybe if we started loading Jest's config, but I doubt it's worth it.

However, you can use `overrides` in `.eslintrc` to enable/disable rules based on file patterns if you want.

**Comment by G-Rath** [comment]
Resolved with the new jest function call parsing logic, which requires functions to either be globals or imported from `@jest/globals`.
